e2e/kubelet/main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+
+	"github.com/sanity-io/litter"
+)
+
+func main() {
+	if err := run(); err != nil {
+		panic(err)
+	}
+}
+
+func run() error {
+	k8sVersion := os.Getenv("KUBE_BINARY_VERSION")
+	binaryPath := fmt.Sprintf("/usr/local/bin/kubelet-%s", k8sVersion)
+	fmt.Println("k8s version is:", k8sVersion)
+	
+	r, w := io.Pipe()
+
+	c1 := exec.Command("sudo", "timeout", "-k", "3", "--preserve-status", "1", binaryPath, "-v", "1", "--container-runtime-endpoint", "unix:///var/run/containerd/containerd.sock")
+	fmt.Println(c1)
+	c1.Stdout = w
+	c1.Stderr = w
+
+	c2 := exec.Command("grep", "FLAG")
+	c2.Stdin = r
+
+	var grepOut bytes.Buffer
+
+	c2.Stdout = &grepOut
+	c2.Stderr = &grepOut
+
+	if err := c2.Start(); err != nil {
+		return fmt.Errorf("failed to start grep pipeline: %q", err)
+	}
+
+	if err := c1.Run(); err != nil {
+		return fmt.Errorf("failed to run kubelet: %q", err)
+	}
+
+	w.Close()
+
+	if err := c2.Wait(); err != nil {
+		fmt.Println(fmt.Errorf("failed to wait for grep to exit: %q", err))
+	}
+
+	flags, err := extractKeyValuePairs(grepOut.Bytes())
+	if err != nil {
+		return fmt.Errorf("failed to extract key value pairs: %q", err)
+	}
+
+	litter.Dump(flags)
```

Comment:
is this going anywhere?

Reply:
it will. this is a 3 stage effort:
1) automate generating flags
2) trigger a PR that will write the flags to a text/json file, when manifest.json is changed on master.
3) use text/json files for validation